### DISCUSSION TITLE
Welcome to bamboo-rs Discussions!

### DISCUSSION THREAD

**Discussion by pietgeursen** [pietgeursen/bamboo-rs#14] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by adzialocha** [comment]
Hello!

Thank you @pietgeursen for writing `bamboo-rs`! We are very happy with it! We're using it as our append-only-log in [p2panda](https://github.com/p2panda/design-document) which is a slowly but steadily growing p2p protocol :smile: here is a small introduction of it at [C36C3](https://media.ccc.de/v/36c3-10756-p2panda).

p2panda will offer a client library written in Rust and TypeScript (which additionally uses a WASM export of the Rust version) and a server "node" implementation as well (can be embedded in a web-view container client, for example in Tauri):

* https://github.com/p2panda/p2panda (Client)
* https://github.com/p2panda/node (Server)

This is an (admittedly confusing, but draft) overview of how we're planning to use Bamboo within p2panda:

![p2panda](https://raw.githubusercontent.com/p2panda/design-document/spec/add-diagrams/spec/assets/data-types.jpg)

:panda_face:

**Reply by pietgeursen** [reply]
This is super cool @adzialocha! Thanks so much for sharing this.

I like how you're using the log_id to split out different feed types. Bamboo is a good fit for that!

**Reply by adzialocha** [reply]
> I like how you're using the log_id to split out different feed types. Bamboo is a good fit for that!

We are also separating system-related logs (`identity` similar to tangles in ssb, `schema` definitions, etc.) in even numbered logs whereas different user feed types are organized in odd numbered ones. It will be helpful in the replication protocol, where we want to prioritize system logs and maybe even give the option to prioritize feed types of schemas which are registered or "whitelisted" on the node.
